esphome/components/web_server/web_server.cpp

```diff
@@ -83,7 +83,7 @@ void WebServer::set_js_include(const char *js_include) { this->js_include_ = js_
 
 void WebServer::setup() {
   ESP_LOGCONFIG(TAG, "Setting up web server...");
-  this->setup_controller();
+  this->setup_controller(true);  // always include internal entities here, as we filter them later
```

Comment:
Why not use this here?
```suggestion
  this->setup_controller(this->include_internal_);
```

Reply:
Using `this->include_internal_` will require reinitialising the entire webserver component when the value changes.

The `Controller::setup_controller` function runs `add_on_state_callback` for each component; however there is no corresponding `remove_on_state_callback` function we can use to un-hook internal entities, if we decide to later stop showing them.

Reply:
> Using this->include_internal_ will require reinitialising the entire webserver component when the value changes.

That's fine, we don't support configuration changes at runtime anyway.